src/inmanta/protocol/methods_v2.py

```diff
@@ -71,6 +72,9 @@ def put_partial(
               * resources: a list of resource objects. Since the version is not known yet resource versions should be set to 0.
               * version_info: Model version information
     :param pip_config: Pip config used by this version
+    :param soft_delete: When True, ignore deletion of resources in removed_resource_sets if they are also being exported
+        (i.e. also present in resource_sets). When False, attempting to delete a resource being exported will raise an
+        exception.
```

Comment:
I think we can save some complexity by making the soft delete purely the responsibility of the exporter, and leave the `put_partial` endpoint untouched. But it might be good to ask the opinion of the rest of the team.